### PR TITLE
Add bindings for creating and extracting GPG signatures

### PR DESCRIPTION
**Pull request by cmr**
These were added in libgit2 v24

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 
 name = "git2"
-version = "0.5.0"
+version = "0.5.1"
```
Comment (alexcrichton) [inline Cargo.toml]: Could this be held off? I prefer to do this in a separate commit.

File: src/repo.rs [hunk 2]
```diff
@@ -819,6 +819,54 @@ impl Repository {
...
+                                                            signature,
+                                                            signature_field));
+            Ok(Binding::from_raw(&raw as *const _))
+        }
+    }
+
+
+    /// Extract the signature from a commit
+    ///
+    /// Returns the signature and the signed data.
```
Comment (alexcrichton) [inline src/repo.rs]: Can this indicate in what order these values are returned?

File: src/repo.rs [hunk 3]
```diff
@@ -819,6 +819,54 @@ impl Repository {
...
+    ///
+    /// Given the unsigned commit object's contents, its signature and the
+    /// header field in which to store the signature, attach the signature to
+    /// the commit and write it into the given repository.
+    ///
+    /// Use `None` in `signature_field` to use the default of `gpgsig`, which is
+    /// almost certainly what you want.
+    ///
+    /// Returns the resulting (signed) commit id.
+    pub fn signed_commit(&self,
```
Comment (alexcrichton) [inline src/repo.rs]: Perhaps this could be called `commit_signed` to mirror the `commit` function above?

**Comment by gchronis** [comment]
Is this issue still relevant? I'd be happy to address your comments and resolve the merge conflicts, but I don't know the conventional way to edit someone else's stale issue. Should I make a PR against @cmr 's branch and hope they merge it? Should I submit a new PR against @alexcrichton 's master branch with the proposed changes, committing as myself?

**Comment by alexcrichton** [comment]
Ah no worries! Ill actually close this for now due to inactivity, but feel free to send a fresh PR to master!
